python/felis/tap_schema.py

```diff
@@ -109,10 +115,68 @@ def __init__(
             self._reflect(engine)
         else:
             self._load_yaml()
+            if extensions_path:
+                self._apply_extensions()
 
         self._create_table_map()
         self._check_tables()
 
+    def _apply_extensions(self) -> None:
+        """Apply extensions from a YAML file to the TAP_SCHEMA schema.
+
+        This method loads extension column definitions from a YAML file and
+        adds them to the appropriate TAP_SCHEMA tables.
+        """
+        if not self.extensions_path:
+            return
+
+        ext_data = None
+
+        if self.extensions_path.startswith("resource://"):
+            ext_resource = ResourcePath(self.extensions_path)
+            ext_content = ext_resource.read()
+            ext_data = yaml.safe_load(ext_content)
+        else:
+            ext_path = Path(self.extensions_path)
+            if not ext_path.exists():
+                raise FileNotFoundError(f"Extensions file not found: {self.extensions_path}")
+            with open(ext_path) as f:
+                ext_data = yaml.safe_load(f)
+
+        if not ext_data or "tables" not in ext_data:
+            logger.warning("Extensions file does not contain 'tables' key, no extensions applied")
+            return
+
+        extension_count = 0
+        for table in self.schema.tables:
+            if table.name not in ext_data["tables"]:
+                continue
+
+            table_extensions = ext_data["tables"][table.name]
+            if not table_extensions:
+                continue
+
+            extension_columns = []
+            for col_def in table_extensions:
+                if "name" not in col_def or "datatype" not in col_def:
```

Comment:
If you load the data into Felis as suggested above then you don't need to check for this, as they are required fields and a validation error would be thrown if they are missing.